### ISSUE TITLE
Add att.canonical to <bibl>

### ISSUE THREAD

**Issue by StefanDumont** [TEIC/TEI#2392]
Since I recently ran (again) into this issue I would like to raise the feature request here:  We use `<bibl>` in our texts for encoding bibliographic references (as [stated in the Guidelines](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/CO.html#COBITY)). The bibliography is hold in an additional TEI file or in Zotero. So we have to link with an URL or ID to an entry in the bibliography. In the case of  `persName`, `placeName`, `orgName`, `name`, `term` and `rs` we can use att.canonical which is very handy (`@key` for local IDs, `@ref` for URIs). Unfortunately `bibl` isn't member of `att.canonical`, so we have to use e.g. `@corresp`  for that purpose. That doesn't seem consistent to me. Especially in processing I notice this often, because we have to treat "bibl-links" extra.

The issue was already discussed multiple times on the TEI-L mailing list, but I didn't see that there have been an issue for it on GitHub. Furthermore, my impression from the TEI-L is that many would support and welcome this enhancement.  

Some discussions on mailing list:
- https://listserv.brown.edu/cgi-bin/wa?A2=ind0902&L=TEI-L&P=R415
- https://listserv.brown.edu/cgi-bin/wa?A2=ind1010&L=TEI-L&D=0&P=50762
- https://listserv.brown.edu/cgi-bin/wa?A2=TEI-L;4db4855c.1103
- https://listserv.brown.edu/cgi-bin/wa?A2=ind2106&L=TEI-L&D=0&P=5694
- https://listserv.brown.edu/cgi-bin/wa?A2=ind2108&L=TEI-L&P=R1530

**Comment by skurzinz** [comment]
I completely agree with the assessment of @StefanDumont, with the same being true for other members of `model.bibLike`. 

Working on #2427 (which adds `<persName>` to the list above through the not perfectly named `att.personal`) I came across all the `(person|place|object|…)Like` models which could also be described as *(named) entities*. Wouldn’t it make sense to sit down and talk about creating a meta class for all these things that can be part of any *ography*? 

IMHO the only thing that’s conceptually different for "works" is that `bibl` and the `bibLike`s refer to abstract bibliographical metadata of a print run/edition and not a concrete manifestation/copy (this one `object type="book"` that Herman Melville wrote his marginalia in), which may complicate things also for the use cases @StefanDumont may be working with, but this is not a showstopper for adding `@key/@ref`.

**Comment by sydb** [comment]
FWIW, I think adding `<bibl>` and its buddies to att.canonical is a good idea.
May need to reinforce that the reference should be to a fuller or canonical _reference_, not to the bibliographic item _itself_.
A possible example from a [web page](https://www.scribbr.co.uk/referencing/harvard-style/) on citation.
~~~xml
<prefixDef ident="b" matchPattern="(\i\c+)" replacementPattern="./bibliography.tei#$1"/>
<!-- ... -->
<p>Referencing is an essential academic skill <bibl type="shrt" ref="b:pears1">Pears and Shields, 2019</bibl>.</p>
<!-- ... in the file ./bibliography.tei: -->
<biblStruct xml:id="pears1">
  <monogr>
    <author>
      <persName>Pears, R.</persName>
    </author>
    <author>
      <persName>Shields, G.</persName>
    </author>
    <title level="m">Cite them right: The essential referencing guide.</title>
    <edition>11th edn.</edition>
    <imprint>
      <date when="2019"/>
      <pubPlace>London</pubPlace>
      <publisher>MacMillan</publisher>
    </imprint>
  </monogr>
</biblStruct>
~~~

**Comment by lb42** [comment]
You will then need to explain why this is or is not preferable to `<ref target='b:pears1'>pears and shields, 2019</ref>`

**Comment by hcayless** [comment]
I'm persuaded that (for example) linking a bibliography to some sort of master bibliography or catalog is a perfectly reasonable thing to want to do. But I agree with @lb42 that we will absolutely need some discussion of when you should use `<bibl>` vs. `<ref>`. My sense is that `@ref` on `<bibl>` is a little bit niche, and that usually you want `<ref>`. That needs to be explained.

**Comment by ebeshero** [comment]
Council F2F at Paderborn: We agree to add `<bibl>` to `att.canonical`, but we need to revise examples and discuss the situation of canonical referencing. We are persuaded that it may be more appropriate to use att.canonical on inline uses of `<bibl>` given the definition of [att.canonical](https://tei-c.org/release/doc/tei-p5-doc/en/html/ref-att.canonical.html). This class of attributes applies to "associate a representation such as a name or title with canonical information about the object being named or referenced", the object being the bibliographic entry. This indeed is the practical usage of inline `<bibl>` referencing described here. We find that `@corresp` does not apply as precisely as the definition of `@ref` to this use case.

We agree that `<ref>` is applicable and recommended for most cases, but not as precise when inline mentions of bibliographic info that want to capture additional information in a short-form reference (with say biblScope information to be tagged). In these cases `<bibl>` is more applicable. Our example in the Guidelines provides `<bibl>` with `@corresp`, but we want to modify this to use `@ref` as more appropriate.

 We need provide examples that apply `<ref>` as well as `<bibl>` with `@ref`. We agree that this will be useful for linked data referencing. 

- [ ] Rearrange Ch. 3.12 to move it earlier in the chapter
- [ ] Discuss and provide examples of when inline `<bibl>` is appropriate, and when you'd use `<ref>` vs. inline `<bibl>`
- [ ] Make clear when you'd want to use `@ref` on `<bibl>`

**Comment by raffazizzi** [comment]
@ebeshero do you recall the reasoning for moving Ch 3.12 earlier in the chapter (3)? That would result in the number changing for a lot of sections – is that wise? What are the benefits?

**Comment by raffazizzi** [comment]
Are we excluding members of `model.biblLike` (`biblFull`, `biblStruct`, `cit`, `event`, `listBibl`, `sourceDesc`, `trait` to name a few) for the time being?

**Comment by sydb** [comment]
`<event>` and `<trait>` are already members of att.canonical. It seems to me `<biblStruct>` is the same, and `<biblFull>` is a similar, sort of thing as `<bibl>`, and thus both should be members of att.canonical. `<sourceDesc>`, `<cit>`, and `<listBibl>` seem entirely different and (IMHO) should _not_ be members of att.canonical.

**Comment by ebeshero** [comment]
@raffazizzi Sorry for this delayed response to your question last week! But on review of proposed revisions to CO, I don't remember why we thought of moving 3.12 earlier in the chapter, and I think it was best to leave it where it is to make these revisions. I think now it may be better to leave that section where people expect to find it, until such time as we want to do larger scale structuring changes to the chapter. (I think that's probably out of scope of this ticket.)

**Comment by ebeshero** [comment]
Completed with https://github.com/TEIC/TEI/pull/2655 !
